src/v3/src/components/Heading/Heading.tsx

```diff
@@ -37,8 +37,8 @@ const Heading: UISchemaElementComponent<{
     >
       <Typography
         id={uischema.id}
-        variant={`h${level ?? 2}`}
-        component={`h${visualLevel ?? 3}`}
+        variant={`h${visualLevel ?? 2}`}
+        component={`h${level ?? 3}`}
```

Comment:
Turns out these were reversed so I've fixed that in this PR and you'll see the updates to transformers (and snapshots) that were rendering this component to reflect that.